sound/soc/sof/trace.c

```diff
@@ -566,12 +572,19 @@ void snd_sof_release_trace(struct snd_sof_dev *sdev)
 			dev_err(sdev->dev, "DMA_TRACE_FREE failed with error: %d\n", ret);
 	}
 
+	if (only_stop) {
+		sdev->dtrace_state = SOF_DTRACE_STOPPED;
```

Comment:
please take previous feedback into account. trace was topped when the IPC to stop is completed succesfully. Please set it to that right state right after the IPC is successful. this if should only have the goto.

Reply:
> please take previous feedback into account. trace was topped when the IPC to stop is completed succesfully. Please set it to that right state right after the IPC is successful. this if should only have the goto.

@ranj063 the trigger stop won't set this flag. And at that time, we don't know which value we should set. At that time, it can be SOF_DTRACE_STOPPED or SOF_Dtrace_DISABLED

Reply:
sorry @libinyang the final state can be DISABLED or STOPPPED depending on what you end up doing. But if the IPC is successful, the flag should be set right after.